### PR TITLE
Add some missing attribute annotations to `trace.pyi`

### PR DESCRIPTION
**Pull request by AlexWaygood**
This will be useful for https://github.com/python/cpython/issues/109413

### CODE REVIEW

File: stdlib/trace.pyi [hunk 1]
```diff
@@ -27,7 +33,21 @@ class CoverageResults:
...
+    inifile: StrPath | None
+    outfile: StrPath | None
+    ignore: _Ignore
+    counts: dict[str, int]
+    pathtobasename: dict[Incomplete, Incomplete]
+    donothing: int
+    trace: int
+    start_time: int | None
+    globaltrace: Callable[[types.FrameType, str, Any], TraceFunction]
+    localtrace: Callable[[types.FrameType, str, Any], TraceFunction]
```
Comment (Avasam) [inline stdlib/trace.pyi:50]: Should those be `TraceFunction` directly?
The main difference is the `TraceFunction | None` return type.

`globaltrace` is passed directly to `threading.settrace` and `sys.settrace` which are both typed to accept a `TraceFunction`

`self.globaltrace` is also assigned a bunch of different possible functions that always return `None` (like `self.globaltrace_trackcallers`, `globaltrace_countfuncs`, etc.)

**Review by Avasam — COMMENTED** [review]
I have two questions, other than that, this looks good assuming `__init__` was already typed correctly\*.

1. (see in-code comment about `TraceFunction`)
2. \*`infile` and `outfile` look like they could work as file descriptors. The online documentation says "name of the file" specifically, but docstring only says "file" https://docs.python.org/3/library/trace.html#trace.Trace I know for the stdlib typeshed sometimes differs from "what's possible" to "do the correct thing", so I'll defer to your judgement here.

File: stdlib/trace.pyi (hunk 1 above)
Comment (AlexWaygood) [reply stdlib/trace.pyi:50]: Good catch. Yep, I think these should both just be `TraceFunction` directly. They're both just used for tracing, so, even if we _could_ be more precise with `self.localstrace`, it's probably fine to simplify here (and the alias gives a better sense of the intended purpose, anyway!)

**Comment by AlexWaygood** [comment]
Thanks for the review! :D

**Pull request merged**